### PR TITLE
Support for custom fog-color texture

### PR DESCRIPTION
**Pull request by pixelflinger**
Fix 282968398

### CODE REVIEW

File: filament/include/filament/Options.h [hunk 1]
```diff
@@ -237,10 +237,32 @@ struct FogOptions {
...
+     * levels as usual. How blurred the base level is somewhat of an artistic decision.
+     *
+     * This simulates a more anisotropic phase-function.
+     *
+     * `fogColorFromIbl` is ignored when skyTexture is specified.
+     *
+     * @see Texture
+     * @see fogColorFromIbl
+     */
+    Texture* skyColor = nullptr;    //!< %codegen_skip_json% %codegen_skip_javascript%
```
Comment (romainguy) [inline filament/include/filament/Options.h:264]: Should we leave a TODO to add JS support later?

File: shaders/src/fog.fs [hunk 2]
```diff
@@ -42,15 +42,15 @@ vec4 fog(vec4 color, highp vec3 view) {
 
 #if MATERIAL_FEATURE_LEVEL > 0
     if (frameUniforms.fogColorFromIbl > 0.0) {
-        // get fog color from envmap
-        // TODO: use a lower resolution mip as we get further (problem we don't have mips!)
-        float lod = frameUniforms.iblRoughnessOneLevel;
+        float normalizedDepth = d * frameUniforms.fogOneOverFarMinusNear - frameUniforms.fogNearOverFarMinusNear;
+        lowp vec2 minMaxMip = unpackHalf2x16(frameUniforms.fogMinMaxMip);
```
Comment (romainguy) [inline shaders/src/fog.fs:46]: lowp isn't needed here, since we default to mediump, therefore fp16.

File: shaders/src/fog.fs [hunk 3]
```diff
@@ -42,15 +42,15 @@ vec4 fog(vec4 color, highp vec3 view) {
...
+        lowp float lod = mix(minMaxMip.y, minMaxMip.x, saturate(normalizedDepth));
```
Comment (romainguy) [inline shaders/src/fog.fs:47]: Same here

**Review by romainguy — APPROVED** [review]

File: filament/include/filament/Options.h (hunk 1 above)
Comment (pixelflinger) [reply filament/include/filament/Options.h:264]: We could, but we already have other places like this (e.g. for the bloom dirt texture). I think the `%codegen_skip_json%` is like a TODO basically.

**Pull request merged**